crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -12,94 +12,145 @@ pub use bevy_ecs_macros::WorldQuery;
 use bevy_ptr::{ThinSlicePtr, UnsafeCellDeref};
 use std::{cell::UnsafeCell, marker::PhantomData};
 
-/// Types that can be queried from a [`World`].
+/// Types that can be used to query a [`World`].
```

Comment:
This introduction is sort of dry. If someone was to happen across this before `Query`, they'd be lost. Providing context as to why this trait exists, or at least link to `Query`  would be a good step forward.  API docs shouldn't have a specific "read-order", and should aggressively link back to core topics so that they're directed there, and dig back deeper if and only if they're required to. `bevy_ecs` is hopelessly pub right now, and `WorldQuery`, while important, is largely an implementation detail with only one outward facing effect: what is and what isn't Queryable.

Reply:
Thanks for the feedback. I used to have some sort of backlink before, but it was like a duplication of what there is in `Query`.

I restored the backlink by slightly changing the first-line short description.